src/sentry/integrations/slack/utils.py

```diff
@@ -423,7 +429,9 @@ def get_channel_id_with_timeout(integration, name, timeout):
     # Look for channel ID
     payload = dict(token_payload, **{"exclude_archived": False, "exclude_members": True})
 
-    if options.get("slack.legacy-app") is True:
```

Comment:
@MeredithAnya You can remove all instances of this setting since this is the only use case of it

Reply:
heck I guess that means removing the docs on it too? Thinking about it again, is there any case that this check wouldn't pass for sentry on-prem?

Reply:
@MeredithAnya  I wouldn't remove it from the docs yet because you'll need it if you have the last version. It doesn't hurt to set it even if we don't use it (we can remove it in the future at some point). This check should work fine for on-prem.